tools/lsp/common.rs

```diff
@@ -99,6 +99,33 @@ pub struct Position {
     pub offset: u32,
 }
 
+/// A versioned file
+#[derive(Clone, Debug, serde::Deserialize, serde::Serialize, PartialEq, Eq)]
+pub struct VersionedPosition {
+    /// The file url
+    url: VersionedUrl,
+    /// The offset in the file pointed to by the `url`
+    offset: u32,
+}
+
+impl VersionedPosition {
+    pub fn new(url: VersionedUrl, offset: u32) -> Self {
+        VersionedPosition { url, offset }
+    }
+
+    pub fn url(&self) -> &Url {
+        self.url.url()
+    }
+
+    pub fn version(&self) -> &UrlVersion {
+        self.url.version()
+    }
+
+    pub fn offset(&self) -> u32 {
+        self.offset
+    }
```

Comment:
IMHO you can just have the fields pub for such small struct. (getters are just useless boiler plate)